src/components/auth/AuthActionButton.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { AccountCircle, MoreVert } from '@material-ui/icons';
+import { Button, useMediaQuery } from '@material-ui/core';
+import { useHistory } from 'react-router-dom';
+
+import { LoadingOverlay } from '@app/contexts/ui/LoadingOverlay';
+import { useModal } from '@app/contexts/ui/Modal';
+import { useSnackbar } from '@app/contexts/ui/Snackbar';
+import { useUser } from '@app/contexts/services/User';
+
+import AuthModal from '../auth/AuthModal';
+import IconButtonMenu from '../common/IconButtonMenu';
+
+import useStyles from './_jss/AuthActionButton.jss';
+
+const AuthActionButton: React.FC = () => {
+  const [isLoading, setLoading] = useState(false);
```

Comment:
nit, just for consistency
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```